src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -84,84 +55,138 @@ OTARequestorInterface * GetRequestorInstance()
+                      response.imageURI.Value().data());
+    }
+    if (response.softwareVersion.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  softwareVersion: %" PRIu32 "", response.softwareVersion.Value());
+    }
+    if (response.softwareVersionString.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  softwareVersionString: %.*s",
+                      static_cast<int>(response.softwareVersionString.Value().size()),
+                      response.softwareVersionString.Value().data());
+    }
+    if (response.updateToken.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  updateToken: %zu", response.updateToken.Value().size());
+    }
+    if (response.userConsentNeeded.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  userConsentNeeded: %d", response.userConsentNeeded.Value());
+    }
+    if (response.metadataForRequestor.HasValue())
+    {
+        ChipLogDetail(SoftwareUpdate, "  metadataForRequestor: %zu", response.metadataForRequestor.Value().size());
+    }
 }
 
-void OnConnectionFailure(void * context, NodeId deviceId, CHIP_ERROR error)
+// TODO: Add unit tests for parsing BDX URI
+CHIP_ERROR OTARequestor::ParseBdxUri(CharSpan uri, NodeId & nodeId, CharSpan & fileDesignator)
 {
-    ChipLogError(SoftwareUpdate, "failed to connect to 0x%" PRIX64 ": %" CHIP_ERROR_FORMAT, deviceId, error.Format());
-}
-// Finds the Requestor instance and calls the corresponding OTARequestor member function
-void OnQueryImageResponse(void * context, const QueryImageResponse::DecodableType & response)
-{
-    OTARequestor * requestorCore = static_cast<OTARequestor *>(GetRequestorInstance());
-
-    assert(requestorCore != nullptr);
+    // Check against minimum length of a valid BDX URI
+    if (uri.size() < kValidBdxUriMinLen)
+    {
+        ChipLogError(SoftwareUpdate, "Expect BDX URI to be at least %d in length, actual length is %zu", kValidBdxUriMinLen,
+                     uri.size());
+        return CHIP_ERROR_INVALID_STRING_LENGTH;
+    }
 
-    requestorCore->mOnQueryImageResponse(context, response);
-}
+    // Check the scheme field matches the BDX prefix
+    CharSpan expectedScheme(bdxPrefix, strlen(bdxPrefix));
+    if (!uri.SubSpan(0, strlen(bdxPrefix)).data_equal(expectedScheme))
+    {
+        ChipLogError(SoftwareUpdate, "Expect scheme to contain prefix %s, actual URI: %.*s", bdxPrefix,
+                     static_cast<int>(uri.size()), uri.data());
+        return CHIP_ERROR_INVALID_SCHEME_PREFIX;
+    }
 
-void OTARequestor::mOnQueryImageResponse(void * context, const QueryImageResponse::DecodableType & response)
-{
-    ChipLogDetail(SoftwareUpdate, "QueryImageResponse responded with action %" PRIu8, response.status);
+    // Extract the node ID from the authority field
+    CharSpan nodeIdString = uri.SubSpan(strlen(bdxPrefix), kNodeIdHexStringMaxLen);
+    uint8_t buffer[kNodeIdHexStringMaxLen];
```

Comment:
It's not a max length, it's the exact length. Perhaps rename `kNodeIdHexStringMaxLen` to `kNodeIdHexStringLen`